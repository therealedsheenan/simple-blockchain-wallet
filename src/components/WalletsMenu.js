import React, { Component } from "react";
import { Link } from "react-router-dom";
import { connect } from "react-redux";
import PropTypes from "prop-types";

import { requestWalletListAction } from "../modules/wallet/actions";
import Placeholder from "./Placeholder";
import loadingSelector from "../modules/loading/selector";

// show list of wallets of the authenticated user
export class WalletsMenu extends Component {
  static defaultProps = {
    walletList: [],
    loading: false
  };

  static propTypes = {
    getWalletListRequest: PropTypes.func.isRequired,
    walletList: PropTypes.array, // eslint-disable-line
    loading: PropTypes.bool
  };

  componentWillMount() {
    this.props.getWalletListRequest();
  }

  render() {
    const { walletList, loading } = this.props;
    return (
      <div className="Wallet-content">
        <h1>Wallet list:</h1>
        {loading ? (
          <Placeholder />
        ) : (
          <div className="Wallet-list">
            {walletList.map(wallet => (
              <div className="Wallet-list--item" key={wallet.id}>
                <Link
                  className="Wallet-list__link"
                  to={`/wallets/${wallet.id}`}
                >
                  {wallet.label}
                </Link>
              </div>
            ))}
          </div>
        )}
      </div>
    );
  }
}

const walletListLoading = loadingSelector(["GET_WALLET_LIST"]);

export default connect(
  state => {
    const { wallet } = state;
    const mapWalletList = Object.keys(wallet.data).map(
      walletId => wallet.data[walletId]
    );
    return {
      walletList: mapWalletList,
      loading: walletListLoading(state)
    };
  },
  dispatch => ({
    getWalletListRequest: () => dispatch(requestWalletListAction())
  })
)(WalletsMenu);
