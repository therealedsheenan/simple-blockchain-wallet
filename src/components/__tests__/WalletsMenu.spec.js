import React from "react";
import { mount } from "enzyme";
import toJson from "enzyme-to-json";
import { BrowserRouter as Router } from "react-router-dom";

import { WalletsMenu } from "../WalletsMenu";

describe("Login component", () => {
  const mockGetWalletListRequest = jest.fn();
  const mockWalletList = [];
  const mockLoading = false;
  const MockWalletsMenu = () => (
    <WalletsMenu
      getWalletListRequest={mockGetWalletListRequest}
      walletList={mockWalletList}
      loading={mockLoading}
    />
  );

  it("Should match snapshot", () => {
    const component = mount(<MockWalletsMenu />);
    expect(toJson(component)).toMatchSnapshot();
  });

  it("Should contain walletList data", () => {
    const mockWalletListData = [
      {
        id: 1,
        label: "label1"
      },
      {
        id: 2,
        label: "label2"
      }
    ];
    const WalletMenuWithRouter = () => (
      <Router>
        <WalletsMenu
          getWalletListRequest={mockGetWalletListRequest}
          walletList={mockWalletListData}
          loading={mockLoading}
        />
      </Router>
    );
    const component = mount(<WalletMenuWithRouter />);
    expect(component.text()).toContain(mockWalletListData[0].label);
    expect(component.text()).toContain(mockWalletListData[1].label);
  });

  // it("Should call componentWillMount", () => {
  //
  //   sinon.spy(WalletsMenu.prototype, "componentWillMount");
  //   expect(WalletsMenu.prototype.componentWillMount.calledOnce).toBeTruthy();
  // });
});
