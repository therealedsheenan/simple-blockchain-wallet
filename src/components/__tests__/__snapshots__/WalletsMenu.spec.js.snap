// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component Should match snapshot 1`] = `
<MockWalletsMenu>
  <WalletsMenu
    getWalletListRequest={[Function]}
    loading={false}
    walletList={Array []}
  >
    <div
      className="Wallet-content"
    >
      <h1>
        Wallet list:
      </h1>
      <div
        className="Wallet-list"
      />
    </div>
  </WalletsMenu>
</MockWalletsMenu>
`;
